Update groovy enforcement label and merge queue documentation (#11897)
DataDog/dd-trace-java: CONTRIBUTING.md

> For reference, the [full list of all labels available](https://github.com/DataDog/dd-trace-java/labels).
> If you feel one is missing, let [the maintainer team](https://github.com/orgs/DataDog/teams/apm-java) know!

### Merge Queue

Pull requests are merged into the protected branches using [the Datadog GitLab merge queue system](https://datadoghq.atlassian.net/wiki/spaces/DEVX/pages/3121612126/MergeQueue).

> Once approved, pull requests can be added to merge queue using the `Merge when ready` button at the bottom of the page, or adding a comment with `/merge` as content.

Reviewer: super small nit, not necessary!

```suggestion
Once approved, pull requests can be added to merge queue using the `Merge when ready` button at the bottom of the page, or by commenting `/merge`.
```

The merge queue will run more tests than the pull request checks, meaning having test failures on merge queue while having pull request checks passing is an expected behavior - the pull request checks are lighter to provide quick feedback during development iterations.

> By default, the merge commit message will be the pull request title.

Reviewer: I thought the default merge commit message was all the PR commit messages squashed into one block 👀  (recent [merge commit](https://github.com/DataDog/dd-trace-java/commit/428b50ed52798b5f08922a32fd9bbfd6355e2b39))

Author: It is... I’ll fix it! Thanks

> A custom commit message can be specified using `/merge --commit-message "custom message"`.

Reviewer: ```suggestion
A custom commit message can be specified by commenting `/merge --commit-message "custom message"`.
```

> Pull request can be forced merged using the following commment as instruction: `/merge -f --reason "reason"`.

Reviewer: ```suggestion
Pull requests can be force-merged by commenting: `/merge -f --reason "reason"`.
```

>[!WARNING]
> By-passing merge queue will effectively merge pull request content without running all continuous integration tests and checks.

> > Only by-pass the merge queue to fix the build or help restoring the continous integration status.

Reviewer: ```suggestion
> Only by-pass the merge queue to fix the build or help restore the continous integration status.
```
